lib/babe/verify.go

```diff
@@ -18,13 +18,210 @@ package babe
 
 import (
 	"encoding/binary"
+	"errors"
 	"fmt"
 
 	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
 )
 
-// verifySlotWinner verifies the claim for a slot, given the types.BabeHeader for that slot.
-func (b *Session) verifySlotWinner(slot uint64, header *types.BabeHeader) (bool, error) {
+// ErrNilNextEpochDescriptor is returned when attempting to get a NextEpochDescriptor that isn't set for an epoch
+var ErrNilNextEpochDescriptor = errors.New("nil NextEpochDescriptor for epoch")
+
+// VerificationManager assists the syncer in keeping track of what epoch is it currently syncing and verifying,
+// as well as keeping track of the NextEpochDesciptor which is required to create a Verifier for an epoch.
+type VerificationManager struct {
+	epochToNextEpochDescriptor map[uint64]*NextEpochDescriptor
+	blockState                 BlockState
+	// TODO: map of epochs to epoch length changes, for use in determining block epoch
+
+	// current epoch information
+	currentEpoch uint64
+	firstBlock   *types.Header // first block of current epoch, may change over course of epoch
+	verifier     *Verifier     // TODO: need to keep historical verifiers, change to map
```

Comment:
I will do it in #662 or after #662 is done, since right now the descriptor is the same for every epoch it's not super relevant. it'll be used for syncing, eg. if we sync up to block 100, but then we find another chain starting at block 20 then we need to sync those past epochs again, so we need to keep the verifier/NextEpochDescriptor for them